### ISSUE TITLE
debug app.js thru chrome inspector

### ISSUE THREAD

**Issue by smolleyes** [jxcore/jxcore-cordova#7]
hi

connected to my set top box thru chrome how can i debug app.js ?
can t see any informations from it :/

thx

**Comment by obastemur** [comment]
Unfortunately you can not debug the inner js application at the moment. This is a roadmap item we have on our list.

**Comment by smolleyes** [comment]
hi

any progress on this ?

thanks :)

**Comment by ktrzeciaknubisa** [comment]
In fact I was able to debug jxcore-cordova app through desktop's Chrome. The trick was to enable it in `MainActivity.java` file. The following link explains how to do it: [Debugging WebViews](https://developer.chrome.com/devtools/docs/remote-debugging). 

So I added the following lines at the end of `onCreate()`:

``` java
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
    WebView.setWebContentsDebuggingEnabled(true);
}
```

This required two more imports:

``` java
import android.os.Bundle;
import android.webkit.WebView;
```

and `minSdkVersion` has to be set to at least to 19 in `AdroidManifest.xml`, e.g.:

``` xml
<uses-sdk android:minSdkVersion="19" android:targetSdkVersion="21" />
```

It also works on AVD.

**Comment by ktrzeciaknubisa** [comment]
An update: looks like after all `minSdkVersion` doesn't need to be set up. It's just that Chrome will not be able to debug WebViews on android devices below 4.4 (KitKat).

**Comment by yaronyg** [comment]
@ktrzeciaknubisa Just to be clear what you did was just debug the webview, not the app.js, correct?

**Comment by ktrzeciaknubisa** [comment]
@yaronyg Correct. That's the whole `index.html` and other scripts loaded by it into webview.

**Comment by PaulosV** [comment]
How far is the app.js debugging on the roadmap? Right now, it would come extremely in handy, as I am trying to debug a nonsense error about a nonexistent file, which goes all the way back to fs.js, and I seem to be unable to catch the exception properly.

**Comment by ktrzeciaknubisa** [comment]
Did you try to catch it by `process.on('uncaughtException')` ?

**Comment by PaulosV** [comment]
I did not, I'll try...
